src/proxy/momento/src/metrics/builder.rs

```diff
@@ -0,0 +1,60 @@
+use goodmetrics::{
+    default_gauge_factory,
+    downstream::{get_client, OpenTelemetryDownstream, OpentelemetryBatcher},
+};
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::sync::mpsc;
+
+use super::{proxy::ProxyMetrics, util::proxy_sum_gauge, RpcMetrics};
+
+pub struct ProxyMetricsBuilder {
+    otel_endpoint: String,
+    batch_interval: Duration,
+    batch_capacity: usize,
+}
+
+impl ProxyMetricsBuilder {
+    pub fn new<S: Into<String>>(otel_endpoint: S) -> Self {
+        Self {
+            otel_endpoint: otel_endpoint.into(),
+            batch_interval: Duration::from_secs(1),
+            batch_capacity: 128,
+        }
+    }
+
+    pub async fn build(self) -> Arc<ProxyMetrics> {
+        let (batch_sender, batch_receiver) = mpsc::channel(self.batch_capacity);
+        let gauge_factory = default_gauge_factory();
+
+        // Set up the OTLP downstream
+        let channel = get_client(
+            self.otel_endpoint.as_str(),
+            || None,
+            goodmetrics::proto::opentelemetry::collector::metrics::v1::metrics_service_client::MetricsServiceClient::with_origin
+        ).expect("connect to otel-collector");
+
+        let otlp_downstream = OpenTelemetryDownstream::new(channel, None);
+        tokio::spawn(async move {
```

Comment:
nit: iirc this async move closure is unnecessary, and same below

Reply:
Fixed in [75e3896](https://github.com/pelikan-io/pelikan/pull/149/commits/75e389667149ebc8a63b945bd0f1d0debed16983)